### ISSUE TITLE
Issue using package.sh on rPi

### ISSUE THREAD

**Issue by mehmetaksakal** [OpenBoard-org/OpenBoard#102]
I'm sure you linux guru type software people will probably know better, I'm an electrical engineer and thus don't know much about this stuff.

Anyway I'm having an issue running package.sh on a raspberry pi, the script borks with the following.

```
-->  Generating control files for package:

dpkg-query: package 'libpaper1' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libx11-6' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5webkit5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5svg5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5multimediawidgets5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5printsupport5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5widgets5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5multimedia5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5gui5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5xmlpatterns5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5network5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5xml5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5script5' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libqt5core5a' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libgles2-mesa' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libgomp1' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
dpkg-query: package 'libgcc-6-dev' is not available
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
-->  Building package:

dpkg-deb: error: parsing file 'debianPackage/DEBIAN/control' near line 10 package 'openboard:armv6l':
 'Depends' field, reference to 'libpaper1': error in version: version string is empty
-->  OpenBoard:
        Package built
```

Maybe it's a good idea to terminate the script on an error. ;)

The problem is that dpkg is unable to get package information for some packages (I do not understand why), thus the version field in the control file is left blank and this the package can't be built.

I've managed to track down the problem to line 351 in the script:

`    echo -n "${tab[$i]} (>= "'dpkg -p ${tab[$i]} | grep "Version: " | awk '{      print $2 }' | sed -e 's/\([:. 0-9?$`

The problem is that `dpkg -p` is used to get the version. I'm not sure how this works on other distros and why the devs are using the `-p` switch because I'm unable to find anything about this switch in the manpages. Some sites suggest the -p (lowercase) stands for purge which isn't what we want. I think the correct switch should be `-s`.

When the line is corrected with the `-s` switch the script runs perfectly.

**Comment by freeheel3r** [comment]
Thanks for reporting this. I believe @kaamui was running into the same errors yesterday, perhaps he can answer this better than I can.

`dpkg -p` doesn't seem to be implemented consistently. For example it works fine on my Ubuntu 14.04 and 16.04 machines, where it shows package status. However, a quick search shows that it may just be an alias for `dpkg-query -p`, but only on some systems. So on others, you'll get these errors.

We should indeed replace it by something that will work more consistently, whether it is `dpkg -s` or `apt-show` or something else.

(And yes, it should terminate when it runs into an error. Thanks for pointing that out!)

**Comment by kaamui** [comment]
I don't think `dpkg` changed between Ubuntu 14.04 and 16.04, as it is a really core function in Debian distros, but maybe the asked repositories are not answering correctly (or at all) anymore.

According to `dpkg` manual :
```
-p, --print-avail package-name...
                  Display details about package-name, as found in
                  /var/lib/dpkg/available. Users of APT-based frontends
                  should use apt-cache show package-name instead.
```

So every APT-based (Debian distros + some others) distros should use apt-cahe show instead of dpkg -p. You'd be careful as apt-cache returns more than one result (probably the current version installed and the minimal requirement for your OS version, maybe ordered systematically the same way but not sure), and use a `| head -1 | ` beetween the `grep "Version: " |` and the `| awk '`, on the line you pointed out.

The fix will be available on the 1.4 release (still have to manage if it works on every supported platforms)

**Comment by mehmetaksakal** [comment]
According to the Ubuntu manpages the -p switch is for the dpkg-query command.
